src/chipsets.h

```diff
@@ -242,13 +243,74 @@ class APA102Controller : public CPixelLEDController<RGB_ORDER> {
 		s0 = (maxBrightness * s0 + (brightness >> 1)) / brightness;
 		s1 = (maxBrightness * s1 + (brightness >> 1)) / brightness;
 		s2 = (maxBrightness * s2 + (brightness >> 1)) / brightness;
+#elif FASTLED_USE_GLOBAL_BRIGHTNESS == 2
+		// Calculate the lightness lookup table for each channel.
+		lutr.init(s0);
+		if (s0 == s1) {
+			lutg.init(lutr);
+		} else {
+			lutg.init(s1);
+		}
+		if (s0 == s2) {
+			lutb.init(lutr);
+		} else if (s1 == s2) {
+			lutb.init(lutg);
+		} else {
+			lutb.init(s2);
+		}
 #else
 		const uint8_t brightness = 0x1F;
 #endif
 
 		startBoundary();
 		while (pixels.has(1)) {
+#if FASTLED_USE_GLOBAL_BRIGHTNESS == 2
+			// On this device, we can get 13 bits of range by using both global 5 bits
+			// PWM and the per channel 8 bits PWM.
+			//
+			// Each channel duty cycle ramps from 100% to 1/(31*255) == 1/7905.
+			//
+			// The goal is to use intensity != 0x1f as much as possible since the
+			// global PWM has a lower frequency than the per channel one.
+			//
+			// To use the full range, we need to discard PixelController's scaling
+			// support since we do not need it.
+			//
+			// Convert each channel to a range between 0 and 7905 using a non-linear
+			// look up table (LUT). This is because the human eye perception is
+			// non-linear.
+			const uint16_t lr = lutr.data[pixels.loadByte0()];
+			const uint16_t lg = lutg.data[pixels.loadByte1()];
+			const uint16_t lb = lutb.data[pixels.loadByte2()];
+			const uint16_t m = lr | lg | lg;
```

Comment:
Typo here: lr | lg | lb, green used twice

Reply:
(!!) thanks

Reply:
I rebased on 3.5.0 and force pushed